server/statistics/buckets/hot_bucket_cache.go

```diff
@@ -0,0 +1,212 @@
+// Copyright 2022 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package buckets
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/log"
+	"github.com/tikv/pd/pkg/keyutil"
+	"github.com/tikv/pd/pkg/logutil"
+	"github.com/tikv/pd/pkg/rangetree"
+	"go.uber.org/zap"
+)
+
+type status int
+
+const (
+	alive status = iota
+	archive
+)
+
+const (
+	// queue is the length of the channel used to send the statistics.
+	queue = 20000
+	// bucketBtreeDegree is the degree of the btree used to store the bucket.
+	bucketBtreeDegree = 10
+
+	// the range of the hot degree should be [-100, 100]
+	minHotDegree = -100
```

Comment:
Does `cold item` need a long time to warm up? and also `hot item` need a long time to be clod?

Reply:
yes, in this sprint, I think the hot degree should be simple to analyze the effect.